File: backend/services/auth/src/utils/apiResponse.ts
import { Response } from "express";

export const sendSuccess = (
  res: Response,
  message: string,
  data: any = null, //data parameter ka type = any , default value = null
  statusCode = 200, //  default status code 200
) => {
  return res.status(statusCode).json({
    success: true,
    message,
    data,
  });
};

export const sendError = (
  res: Response,
  message: string,
  data: any = null,
  statusCode = 400,
) => {
  return res.status(statusCode).json({
    success: false,
    message,
    data,
  });
};

File: backend/services/chat/utils/api_response.py
from flask import jsonify


def send_success(message: str, data=None, status_code: int = 200):
    return (
        jsonify({
            "success": True,
            "message": message,
            "data": data,
        }),
        status_code,
    )


def send_error(message: str, data=None, status_code: int = 500):
    return (
        jsonify({
            "success": False,
            "message": message,
            "data": data,
        }),
        status_code,
    )